runners/core-java/src/main/java/org/apache/beam/runners/core/StepContext.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.core;
+
+/**
+ * Per-step, per-key context used for retrieving state.
```

Comment:
Maybe improve javadoc and pull in some from the interface previously known as ExecutionContext?

The context in which a specific step is executing. This interface should be implemented by runners to provide the necessary hooks for implementing state and timers.

This is guaranteed to exist during processing, but does not necessarily persist between different batches of work.